dvc/ignore.py

```diff
@@ -27,7 +27,22 @@ def __init__(self, ignore_file_path, tree):
         self.dirname = os.path.normpath(os.path.dirname(ignore_file_path))
 
         with tree.open(ignore_file_path, encoding="utf-8") as fobj:
-            self.ignore_spec = PathSpec.from_lines(GitWildMatchPattern, fobj)
+            path_spec_lines = fobj.readlines()
+            regex_pattern_list = list(
+                map(GitWildMatchPattern.pattern_to_regex, path_spec_lines)
+            )
+        ignore_pattern_list = [i for i, j in regex_pattern_list if j is True]
+        include_pattern_list = [i for i, j in regex_pattern_list if j is False]
+        self.ignore_spec = (
+            re.compile("|".join(ignore_pattern_list))
+            if ignore_pattern_list
+            else None
+        )
+        self.include_spec = (
+            re.compile("|".join(include_pattern_list))
+            if include_pattern_list
+            else None
+        )
```

Comment:
Aren't we loosing order here? I think that matters.

Reply:
> @karajan1001 Windows build is failing.
Sorry for late, I went to the hospital last weekend but had recovered now.


> Aren't we loosing order here? I think that matters.

Yes，It losing order here。According to [gitignore](https://git-scm.com/docs/gitignore)
`
An optional prefix "!" which negates the pattern; any matching file excluded by a previous pattern will become included again. 
`
Only late rules can override early ones.